narwhal/executor/src/lib.rs

```diff
@@ -49,6 +49,16 @@ pub trait ExecutionState {
         transaction: Vec<u8>,
     );
 
+    /// Notifies executor that narwhal commit boundary was reached
+    /// Consumers can use this boundary as an approximate signal that it might take some
+    /// time before more transactions will arrive
+    /// Consumers can use this boundary, for example, to form checkpoints
+    ///
+    /// Current implementation sends this notification at the end of narwhal certificate
+    ///
+    /// In the future this will be triggered on the actual commit boundary, once per narwhal commit
+    async fn notify_commit_boundary(&self, _consensus_output: &Arc<ConsensusOutput>) {}
```

Comment:
I am not sure I understand the future plan. Is the plan to implement this function later?

Reply:
Well, yes, it will be implemented in sui handler.

There are details in the [RFC](https://github.com/MystenLabs/sui/issues/5763) on how this will be used, but TLDR is that validators will create a checkpoint here.

For now this is cert boundary since it is easy to get, but eventually it will be changed to be an actual narwhal commit boundary